eth/transaction_from_raw.go

```diff
@@ -148,6 +152,108 @@ func (t *Transaction) FromRaw(input string) error {
 		t.Hash = raw.Hash()
 		t.From = *sender
 		return nil
+	case firstByte == byte(TransactionTypeBlob):
+		// EIP-4844 transaction
+		var (
+			body        rlp.Value
+			blobs       []Data
+			commitments []Data
+			proofs      []Data
+		)
+		// 0x03 || rlp([tx_payload_body, blobs, commitments, proofs])
+		payload := "0x" + input[4:]
+		if err := rlpDecodeList(payload, &body, &blobs, &commitments, &proofs); err != nil {
+			return err
+		}
+		if len(body.List) != 14 {
+			return errors.New("blob transaction invalid tx body length")
+		}
+		// TransactionPayloadBody is itself an RLP list of:
+		// [chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, to, value, data, access_list, max_fee_per_blob_gas, blob_versioned_hashes, y_parity, r, s]
+		if err := rlpDecodeList(&body, &chainId, &nonce, &maxPriorityFeePerGas, &maxFeePerGas, &gasLimit, &to, &value, &data, &accessList, &maxFeePerBlobGas, &blobVersionedHashes, &v, &r, &s); err != nil {
+			return errors.Wrap(err, "could not decode RLP components")
+		}
+
+		if len(blobVersionedHashes) != len(blobs) || len(blobs) != len(commitments) || len(commitments) != len(proofs) {
+			return errors.New("mismatched blob field counts")
+		}
+
+		// TODO: at this point we could verify these two constraints
+		//
+		// - blobVersionedHashes[i] = "0x01" + sha256(commitments[i])[4:]
+		// - the KZG commitments match the corresponding blobs and proofs
+		//
+		// However, this all requires additionally implementing a sha256 method
+		// for eth.Data and the use of a KZG proof framework, both of which
+		// feel outside the scope of this package, especially considering
+		// that these fields are not exposed at the JSONRPC layer which is
+		// our primary focus.
+		//
+		// In pseudocode this all would look something like:
+		//
+		//  for i := range blobVersionedHashes {
+		//	  blobHash := commitments[i].Sha256()
+		//	  versionedHash := "0x01" + blobHash[4:]
+		//	  if blobVersionedHashes[i] != versionedHash {
+		//	    return errors.New("incorrect blob versioned hash")
+		//	  }
+		//	  if err := kzg_verify_proofs(commitments[i], blobs[i], proofs[i]); err != nil {
+		//	    return err
+		//	  }
+		//  }
+		//
+
+		if r.Int64() == 0 && s.Int64() == 0 {
+			return errors.New("unsigned transactions not supported")
+		}
+
+		t.Type = OptionalQuantityFromInt(int(firstByte))
+		t.ChainId = &chainId
+		t.Nonce = nonce
+		t.MaxPriorityFeePerGas = &maxPriorityFeePerGas
+		t.MaxFeePerGas = &maxFeePerGas
+		t.Gas = gasLimit
+		t.To = to
+		t.Value = value
+		t.Input = data
+		t.AccessList = &accessList
+		t.MaxFeePerBlobGas = &maxFeePerBlobGas
+		t.BlobVersionedHashes = blobVersionedHashes
+		t.V = v
+		t.YParity = &v
+		t.R = r
+		t.S = s
+
+		t.BlobBundle = &BlobsBundleV1{
+			Blobs:       blobs,
+			Commitments: commitments,
+			Proofs:      proofs,
+		}
+
+		signature, err := NewEIP155Signature(r, s, v)
```

Comment:
Ooops! This should be:

```suggestion
		signature, err := NewEIP2718Signature(chainId, r, s, v)
```

😊 I have a couple more unit tests incoming to confirm this.

Reply:
Fixed as part of https://github.com/INFURA/go-ethlibs/commit/9e026b64ff67dc6c4cba6c4deaca39d0f1587119